### PR TITLE
Add HttpRange to azure-core

### LINKED ISSUE #21791: Add HttpRange
Add a new class into `azure-core`'s HTTP package named `HttpRange`. The API for this new model should be the following:

```java
public final class HttpRange {
  public HttpRange(long offset);
  public HttpRange(long offset, long length);
}
```

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #21791 

This PR adds `HttpRange` to `azure-core` as a model class to support performing `Range` based HTTP requests.

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Objects;
+
+/**
+ * Represents a range of bytes within an HTTP resource.
+ * <p>
+ * The range starts at the {@link #getOffset() offset} inclusively and ends at {@link #getOffset() offset} + {@link
+ * #getLength() legnth} exclusively, or offset + length - 1.
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java:14]:
```suggestion
 * #getLength() length} exclusively, or offset + length - 1.
```

File: sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ * <p>
+ * If {@link #getLength() length} is unspecified, null, then the range extends to the end of the HTTP resource.
+ */
+public final class HttpRange {
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java:18]: Should this go into core-experimental first to vet out that it works for ACS with their download APIs ?

Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java:18]: This seems to be generic enough to be protocol agnostic. Should we have `Http` in the name?

Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java:18]: While the functionality seems generic, the toString returning `bytes=<offset>-<offset + length>` is tightly tied to what the `Range` header expected in HTTP.

Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java:18]: I can move this into experimental for now

Comment (srnagar) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java:18]: Does `toString()` have to format it in a way that it is tied to Http?

Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java:18]: Having `toString` be `HTTP` specific isn't a strong requirement. The class could be made into a more generic `Range` and utility methods could be provided to format the range into an HTTP `Range` header.

Comment (srnagar) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRange.java:18]: Let's leave it as HttpRange for now as there's no other usecase that we know of that required `Range` type.

**Review by srnagar — APPROVED** [review]

**Pull request merged**